### PR TITLE
Boot firmware update fixes

### PR DESCRIPTION
**Pull request by ricardosalveti**

### CODE REVIEW

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in [hunk 1]
```diff
@@ -37,6 +33,12 @@ if fiovb init ${devnum} && test -n "${board_is_closed}"; then
 	setenv fiovb_rpmb 1;
 else
 	echo "${fio_msg} Using ubootenv"
+	# Make sure basic environment is valid and call saveenv if not yet set
+	if test -z "${bootcount}"; then setenv bootcount 0; fi
+	if test -z "${rollback}"; then setenv rollback 0; fi
+	if test -z "${upgrade_available}"; then setenv upgrade_available 0; fi
+	if test -z "${bootupgrade_available}"; then setenv bootupgrade_available 0; run saveenv_mmc; fi
```
Comment (Tim-Anderson) [inline meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in:40]: this only saves the environment of bootupgrade_available was not set, does not address the other 3 variables

**Review by Tim-Anderson — COMMENTED** [review]
saveenv_mmc comment

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in:40]: Yeah, my idea was to use one variable as trigger, as we shouldn't be saving 4 times here (and bootcount could already be set to 0).

I decided to use bootupgrade_available as that is the only one that is "ours".

**Comment by Tim-Anderson** [comment]
My suggestion would be to setenv save 0; if any test -z then setenv save 1
and at the end test if save is set and then run saveenv_mmc

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in:40]: From what I saw only rollback and bootupgrade_available are not set by u-boot (when comparing with machines setting their own standard / hardcoded environment and boot logic).

**Comment by ricardosalveti** [comment]
> My suggestion would be to setenv save 0; if any test -z then setenv save 1 and at the end test if save is set and then run saveenv_mmc

Updated as suggested.

**Review by igoropaniuk — APPROVED** [review]
LGTM

**Pull request merged**